soroban-sdk/src/testutils.rs

```diff
@@ -67,3 +70,24 @@ pub trait Accounts {
     /// Remove an account.
     fn remove(&self, id: &AccountId);
 }
+
+/// Generates a Rust array of N random bytes.
+pub fn random_bytes_array<const N: usize>() -> [u8; N] {
+    let mut arr = [0u8; N];
+    rand::thread_rng().fill_bytes(&mut arr);
+    arr
+}
+
+/// Generates a random Stellar `AccountId` XDR.
+pub fn random_account_id() -> xdr::AccountId {
+    xdr::AccountId(xdr::PublicKey::PublicKeyTypeEd25519(xdr::Uint256(
+        random_bytes_array(),
+    )))
+}
+
+impl Env {
+    /// Generates a Host-owned array of `N` random bytes.
+    pub fn random_bytes<const N: usize>(&self) -> BytesN<N> {
+        BytesN::from_array(self, &random_bytes_array())
```

Comment:
I just realized there's a couple things inconsistent with the placement of these functions with the way other test utilities are structured, so I'm going to follow up with addressing that. The issues are:
- One of the functions is exposing XDR types, but we try and expose only SDK types if we can. All SDK types are convertible to XDR types anyway, and it prevents users getting confused about what types they can or can't use in different situations.
- We avoid putting functions like "random account id" on Env, and put them on the type itself, so that the Env isn't like an everything object of sorts.

Reply:
Follow up is in https://github.com/stellar/rs-soroban-sdk/pull/767.